docs/errors/E0717.md

```diff
@@ -0,0 +1,46 @@
+# E0717: using a '.' after a '?.' might fail, since '?.' might return 'undefined'
+
+```config-for-examples
+{
+  "globals": {
+  }
+}
+```
+
+In Javascript, the `?.` operator performs a conditional access only if the left-hand side is not `null` or `undefined`. If the left-hand side is `null` or `undefined`, the expression returns `undefined` and the right-hand side is not evaluated. Using it suppresses the error that would otherwise be thrown if the left-hand side were `null` or `undefined` and the operator `.` were used instead. However, if the left-hand side is not `null` or `undefined`, the expression returns the value of the right-hand side, which may be `null` or `undefined`. If the right-hand side is `null` or `undefined`, using the `.` operator will throw an error:
+
+```js
+let a = { b: 1 };
+let b = a?.b; // ok
+let c = a?.b?.a; // ok
+let d = a?.b?.a.c; // throws an error
+```
+
+In the example above, the last statement will throw an error, because `a.b` is not `null` or `undefined`, but `a.b.a` is `undefined`, and `undefined.c` throws an error.
+
+This also affects optional chaining with function calls (`?.()`):
+
+```js
```

Comment:
Must fix: Our docs require one wrong/invalid example followed by any number of correct examples. Here you have more than one wrong/invalid example. (This is machine-checked by the [failing "check error docs" CI job](https://github.com/quick-lint/quick-lint-js/actions/runs/7307489068/job/19971420944?pr=1133).)

I think the second example (line 23) and third example (line 34) don't add much value. The point of the error docs (in my mind) isn't to cover every case, but instead to help the reader confirm the problem then figure out a solution.

I think the fix here is to remove the extra examples.